File: 2018/day01-2/src/main.jl
input = open("../input/input.txt")
freqs = read(input, String)

total = 0
seen = Set([0])
twice = -1

while twice <= -1
    for f in split(freqs)
        num = parse.(Int, f)
        global total += num

        if total in seen
            global twice = total
            break
        end

        push!(seen, total)
    end
end

println(twice)

close(input)

File: 2018/day01-2/src/main.rs
use std::{
    collections::HashSet,
    error::Error,
    io::{ Read, stdin }
};

fn main() -> Result<(), Box<Error>> {
    let mut total = 0;

    let mut seen = HashSet::new();
    seen.insert(0);

    let mut input = String::new();
    stdin().read_to_string(&mut input)?;

    loop {
        for i in input.lines() {
            let num: i32 = i.parse()?;

            total += num;

            if seen.contains(&total) {
                println!("{}", total);
                return Ok(());
            }

            seen.insert(total);
        }
    }
}
